homeassistant/components/sql/__init__.py

```diff
@@ -37,8 +37,10 @@
 
 
 def validate_sql_select(value: str) -> str:
-    """Validate that value is a SQL SELECT query."""
-    if not value.lstrip().lower().startswith("select"):
+    """Validate that value is a SQL SELECT or WITH ... SELECT (a SELECT using a Common Table Expression) query."""
+    if not value.lstrip().lower().startswith(
+        "select"
+    ) and not value.lstrip().lower().startswith("with"):
```

Comment:
```suggestion
    if not value.lstrip().lower().startswith(("select", "with")):
```

Reply:
This could also use a test that has a CTE expression

Reply:
This also needs to be updated in `config_flow.py` also at least

Reply:
Updated the PR!